packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1215,13 +1214,14 @@ new PrismaClient({
       })
     }
 
-    private async _getDmmf(params: Pick<InternalRequestParams, 'clientMethod' | 'callsite'>) {
+    private _getDmmf = callOnce(async (params: Pick<InternalRequestParams, 'clientMethod' | 'callsite'>) => {
       try {
-        return await this._engine.getDmmf()
+        const dmmf = await this._engine.getDmmf()
+        return new DMMFHelper(getPrismaClientDMMF(dmmf))
```

Comment:
```suggestion
        if(this._dmmf) { return this_dmmf }
        const dmmfString = await this._engine.getDmmf();
        this._dmmf = new DMMFHelper(getPrismaClientDMMF(dmmfString));
        return this_dmmf
```

Would this be simpler, just set and cache a property on the class?

Reply:
Code calling code already doing this. The problem is, there is nothing cached while the promise is in flight, so if 500 queries happen in the same tick like in original issue `getDMMF` will be called 500 time

Reply:
I see, I do believe in the codebase we already have a solution for this and that's returning the cached promise see: 

https://github.com/prisma/prisma/blob/main/packages/engine-core/src/library/LibraryEngine.ts#L321

But I do like your helper so maybe we could utilize it in another PR there too.

Reply:
Yea, I went with helper exactly because I saw we are doing it in several places already and we might want to pull common code out and reuse it.